app/questionnaire/questionnaire_schema.py

```diff
@@ -1033,15 +1034,36 @@ def get_answer_format_for_calculated_summary(
         """
         # Type ignore: the block will exist for any valid calculated summary id
         calculated_summary_block: ImmutableDict = self.get_block(calculated_summary_block_id)  # type: ignore
-        first_answer_id = get_calculated_summary_answer_ids(calculated_summary_block)[0]
+        answer_ids = get_calculated_summary_answer_ids(calculated_summary_block)
+        decimal_limit = self.get_decimal_limit(answer_ids)
+        first_answer_id = answer_ids[0]
         first_answer = self.get_answers_by_answer_id(first_answer_id)[0]
         return {
             "type": first_answer["type"].lower(),
             "unit": first_answer.get("unit"),
             "unit_length": first_answer.get("unit_length"),
             "currency": first_answer.get("currency"),
+            "decimal_places": decimal_limit,
         }
 
+    def get_grand_calculated_summary_decimal_limit(
+        self, calculated_summary_ids: list[str]
```

Comment:
I'd favour this being in `grand_calculated_summary_context.py` over here since its only used there and specific to it (maybe it could be inline as well) but this does fix the issue 🙌

Reply:
@MebinAbraham as we discussed I didn't do this but changed the name to `get_decimal_limit_from_sources` with a minor refactor - was this what you had in mind?

Reply:
I was thinking sources as in the values sources itself, since this only handles calc summary and passed in ids instead of sources it should be specific to calc summary, like `get_decimal_limit_from_calculated_summaries`

Reply:
Ahh ok - renamed in the interests of getting it across 👍